src/redux/state.js

```diff
@@ -1,82 +1,96 @@
-import { rerenderEntireTree } from '../render';
+const store = {
 
-const state = {
-  profilePage: {
-    posts: [
-      {
-        id: '1',
-        message: 'Hi, how are you',
-        likesCount: '12',
-      },
-      {
-        id: '2',
-        message: 'It`s my first posts',
-        likesCount: '7',
-      },
-    ],
-    newPostText: '',
+  _state: {
+    profilePage: {
+      posts: [
+        {
+          id: '1',
+          message: 'Hi, how are you',
+          likesCount: '12',
+        },
+        {
+          id: '2',
+          message: 'It`s my first posts',
+          likesCount: '7',
+        },
+      ],
+
+      newPostText: '',
+    },
+
+    dialogsPage: {
+      dialogs: [
+        {
+          id: '1',
+          name: '105',
+        },
+        {
+          id: '2',
+          name: 'FrinteZz',
+        },
+        {
+          id: '3',
+          name: 'Gaming',
+        },
+        {
+          id: '4',
+          name: 'Pominki',
+        },
+        {
+          id: '5',
+          name: 'Side',
+        },
+        {
+          id: '6',
+          name: 'EFsquad',
+        },
+      ],
+
+      messages: [
+        {
+          id: '1',
+          message: 'qq all',
+        },
+        {
+          id: '2',
+          message: 'hello world',
+        },
+        {
+          id: '3',
+          message: 'priveti4 ot Marmeladi4a',
+        },
+      ],
+    },
   },
-  dialogsPage: {
-    dialogs: [
-      {
-        id: '1',
-        name: '105',
-      },
-      {
-        id: '2',
-        name: 'FrinteZz',
-      },
-      {
-        id: '3',
-        name: 'Gaming',
-      },
-      {
-        id: '4',
-        name: 'Pominki',
-      },
-      {
-        id: '5',
-        name: 'Side',
-      },
-      {
-        id: '6',
-        name: 'EFsquad',
-      },
-    ],
 
-    messages: [
-      {
-        id: '1',
-        message: 'qq all',
-      },
-      {
-        id: '2',
-        message: 'hello world',
-      },
-      {
-        id: '3',
-        message: 'priveti4 ot Marmeladi4a',
-      },
-    ],
+  getState() {
+    return this._state;
+  },
+
+  _callSubscriber() {
+    console.log('changed');
   },
-};
 
-export const addPost = () => {
-  const newPost = {
-    id: 7,
-    message: state.profilePage.newPostText,
-    likesCount: 0,
-  };
+  subscribe(observer) {
+    this._callSubscriber = observer;
+  },
 
-  state.profilePage.posts.push(newPost);
-  state.profilePage.newPostText = '';
-  rerenderEntireTree(state);
-};
+  dispatch(action) {
+    if (action.type === 'ADD-POST') {
+      const newPost = {
+        id: 7,
+        message: this._state.profilePage.newPostText,
```

Comment:
All these "_name" variables with the "_" in their naming should have triggered eslint - since there's no need to mark the variable in such a way, they're still public (if you're not implementing the export interface).

Reply:
I added constants for the action type

Reply:
I think I forgot to type the "'name' variables with the '' in their naming" properly.
What I suggest is that you don't you _state or _callSubscriber as it is not considered as "private properties" by V8.
And therefore, eslint should mark it as bad practice.